src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -76,20 +84,49 @@ public EvaluationContext getEvaluationContext() {
      * {@inheritDoc}
      */
     public void setProvider(FeatureProvider provider) {
+        if (provider == null) {
+            this.providers.remove(DEFAULT_PROVIDER_KEY);
```

Comment:
any reason for removing the entry from map only for default client and not for named client when the given provider is null? If the intention is remove it from map, we should move this logic inside the `setProvider(String clientName, FeatureProvider provider)`